### PR TITLE
Futurev6

### PR DESCRIPTION
**Pull request by lacan**
Improves support for Operetta Phoenix. 

More importantly, it allows for opening images saved with the 6th verion of the Operetta data export, which is not compatible with the previous versions. 

This V6 update of the Operetta effectively breaks the olr old pure Groovy Operetta importer Script... 

Infitnite thanks to @NicoKiaru for the byte to float to short conversion help!

### CODE REVIEW

File: src/main/java/ch/epfl/biop/operetta/OperettaManager.java [hunk 1]
```diff
@@ -515,6 +524,8 @@ public ImageStack readSingleStack(WellSample field, final int downscale, HyperRa
         StopWatch sw = new StopWatch();
         sw.start();
 
+        OMEXMLMetadataRoot meta = (OMEXMLMetadataRoot) metadata.getRoot();
```
Comment (lacan) [inline src/main/java/ch/epfl/biop/operetta/OperettaManager.java:527]: No need

File: src/main/java/ch/epfl/biop/operetta/OperettaManager.java [hunk 2]
```diff
@@ -563,6 +573,54 @@ public ImageStack readSingleStack(WellSample field, final int downscale, HyperRa
         return stack;
     }
 
+    private ImageProcessor openTiffFileAsImageProcessor(String imagePath)  {
```
Comment (lacan) [inline src/main/java/ch/epfl/biop/operetta/OperettaManager.java:576]: Javadoc

File: src/main/java/ch/epfl/biop/operetta/OperettaManager.java [hunk 3]
```diff
@@ -1220,6 +1278,19 @@ public static class Builder {
 
         private IFormatReader reader = null;
 
+        private boolean flip_horizontal = false;
+        private boolean flip_vertical = false;
+
+        public Builder flipHorizontal(boolean isFlip) {
```
Comment (lacan) [inline src/main/java/ch/epfl/biop/operetta/OperettaManager.java:1284]: Javadoc

File: src/main/java/ch/epfl/biop/operetta/OperettaManager.java [hunk 4]
```diff
@@ -1220,6 +1278,19 @@ public static class Builder {
...
+            this.flip_horizontal = isFlip;
+            return this;
+        }
+
+        public Builder flipVertical(boolean isFlip) {
```
Comment (lacan) [inline src/main/java/ch/epfl/biop/operetta/OperettaManager.java:1289]: javadoc

File: src/main/java/ch/epfl/biop/operetta/commands/OperettaImporter.java [hunk 5]
```diff
@@ -53,27 +53,46 @@ public class OperettaImporter implements Command {
     @Parameter
     CommandService cs;
 
+    private enum XMLFILE {
```
Comment (lacan) [inline src/main/java/ch/epfl/biop/operetta/commands/OperettaImporter.java:56]: Javadoc. Consider adding original '_flex' index as well

File: src/main/java/ch/epfl/biop/operetta/commands/OperettaImporter.java [hunk 6]
```diff
@@ -53,27 +53,46 @@ public class OperettaImporter implements Command {
...
+            this.indexFileName = indexFileName;
+            this.description = description;
+        }
+        private String getIndexFileName() { return this.indexFileName; }
+        public String getDescription() { return this.description; }
+    }
+
     @Override
     public void run() {
+        File f = null;
```
Comment (lacan) [inline src/main/java/ch/epfl/biop/operetta/commands/OperettaImporter.java:73]: Change the way candidates are working here to account for the memo file part

Re-add memoization check part

File: src/main/java/ch/epfl/biop/operetta/commands/OperettaImporter.java [hunk 7]
```diff
@@ -53,27 +53,46 @@ public class OperettaImporter implements Command {
...
+            for (XMLFILE version : XMLFILE.values()) {
+                IJ.log("\t" + version.getIndexFileName() + " (" + version.getDescription() + ")");
+            }
             return;
         }
 
         int sizeInMb = (int) ((double) FileUtils.sizeOf(f) / (double) (1024 * 1024));
         IJ.log("- Opening Operetta dataset " + f.getAbsolutePath() + " (" + sizeInMb + " Mb)");
 
-        File fmemo = new File(folder, ".Index.idx.xml.bfmemo");
```
Comment (lacan) [inline src/main/java/ch/epfl/biop/operetta/commands/OperettaImporter.java:68]: Re add this

**Review by NicoKiaru — APPROVED** [review]
LGTM

**Pull request merged**